### PR TITLE
Convert CI unit tests to use docker images.

### PR DESCRIPTION
**Pull request by jakelandis**
Alternative to https://github.com/logstash-plugins/logstash-filter-grok/pull/139 for discussion.

Run unit tests from within docker containers via Docker compose. 

Similar to https://github.com/logstash-plugins/logstash-filter-grok/pull/139:

- uses snapshots instead of building logstash
- relies only on the ruby that ships with logstash
- speeds test suite from 8 minutes to ~ 2 minutes

Dis-similar to https://github.com/logstash-plugins/logstash-filter-grok/pull/139
- tests run inside docker container 
- doesn't require any local setup other then docker
- runs with docker compose to match an integration testing setup (e.g. one way to do unit or integration tests)
- arguably more complicated then https://github.com/logstash-plugins/logstash-filter-grok/pull/139
- can not take advantage of Travis' bundler cache.

### CODE REVIEW

**Comment by jakelandis** [comment]
This is what a unit/integration testing pair could look like, both using docker. 

https://github.com/jakelandis/logstash-input-azure_event_hubs/tree/re-implement-with-event-processor-host/ci

To run the unit tests 
`ci/unit/docker-test.sh`
To run the integration tests
`ci/integration/docker-test.sh`

(note the above example uses a manual integration testing method, but can be evolved to a pure automated test without changing the general structure).

**Comment by webmat** [comment]
I really love the ability to run the integration tests reliably locally. If they fail, they will also fail in Travis, and they need to be fixed.

You're sure not to be missing any special quirk (e.g. "don't run ElasticSearch when you run the output-es tests").

File: ci/unit/command.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# This is intended to be run inside the docker container as the command of the docker-compose.
+set -ex
+
+export PATH=$PATH:$PWD/vendor/jruby/bin
+gem install bundler
+cd ../plugins/this
+cp /usr/share/logstash/logstash-core/versions-gem-copy.yml /usr/share/logstash/logstash-core-plugin-api/versions-gem-copy.yml
+#./gradlew vendor
+bundle install
+bundle exec rspec -fd --pattern spec/**/*_spec.rb,spec/**/*_specs.rb
```
Comment (jsvd) [inline ci/unit/command.sh]: why the need to include `_specs.rb`? we could just have `bundle exec rspec -fd`

File: .travis.yml [hunk 2]
```diff
@@ -1,18 +1,17 @@
 ---
-sudo: false
+sudo: required
+services: docker
+addons:
+  apt:
+    packages:
+      - docker-ce
 language: ruby
```
Comment (jsvd) [inline .travis.yml]: we don't need to state `language: ruby` anymore, I guess?

File: ci/unit/command.sh (hunk 1 above)
Comment (jakelandis) [reply ci/unit/command.sh]: the default pattern is '"**{,/*/**}/*_spec.rb"' , and i _think_ i saw (wrote?) some tests that end in _specs. Also if we introduce more integration specs they are less likely to get picked up by the default glob.

File: .travis.yml (hunk 2 above)
Comment (jakelandis) [reply .travis.yml]: yeah, i don't think this is needed anymore.

**Comment by jakelandis** [comment]
closing this PR as stale

**Pull request closed without merging**